includes/wc-formatting-functions.php

```diff
@@ -260,7 +260,7 @@ function wc_format_decimal( $number, $dp = false, $trim_zeros = false ) {
 	$decimals = array( wc_get_price_decimal_separator(), $locale['decimal_point'], $locale['mon_decimal_point'] );
 
 	// Remove locale from string.
-	if ( ! is_float( $number ) ) {
+	if (  gettype( $number ) === 'string' && ! is_numeric( $number ) ) {
```

Comment:
It's just for hardening the function. Think about e.g. integers (no much relevant here, because str/preg_replace will not do anything on integers, may only return a wrong variable type php-notice.
This part of code will really be relavant if you guys want to readd the thousands removal snippet.